compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp

```diff
@@ -143,38 +143,24 @@ static TuningSpecsToMerge collectTuningSpecsToMerge(ModuleOp module) {
 }
 
 // Renames a `NamedSequenceOp` to resolve name conflicts caused by merging
-// tuning specs.
-// The name conflict resolution strategy follows below two rules:
-//   1. If the `NamedSequenceOp` is inside a module with a valid symbol name,
-//      its new name is prefixed with its containing module's symbol name.
-//   2. If the module has no symbol name, an incrementing counter is used
-//      to generate a unique prefix (e.g., `m0_`, `m1_`, etc.).
+// tuning specs by appending a numeric suffix until a unique name is found.
 static void updateNamedSequenceOp(
     NamedSequenceOp op, OpBuilder &builder,
     llvm::DenseMap<NamedSequenceOp, ForeachMatchOp> &namedSequenceToUser,
-    llvm::DenseMap<ModuleOp, std::string> &unnamedModuleNames,
-    unsigned &unnamedModuleCounter) {
+    llvm::DenseSet<StringRef> &seenNames) {
   StringRef specName = op.getSymName();
-  ModuleOp parentModule = op->getParentOfType<ModuleOp>();
-  assert(parentModule);
-  StringAttr parentSymbol = parentModule.getSymNameAttr();
-  std::string moduleName;
-  if (parentSymbol) {
-    moduleName = parentSymbol.getValue().str();
-  } else {
-    if (unnamedModuleNames.contains(parentModule)) {
-      moduleName = unnamedModuleNames[parentModule];
-    } else {
-      std::string newModuleName =
-          llvm::formatv("m{}", unnamedModuleCounter).str();
-      ++unnamedModuleCounter;
-      unnamedModuleNames[parentModule] = newModuleName;
-      moduleName = newModuleName;
-    }
+
+  // Ensure the name is unique by appending a numeric suffix if needed.
+  std::string uniqueNewSpecName = specName.str();
+  unsigned suffix = 0;
+  while (seenNames.contains(uniqueNewSpecName)) {
+    uniqueNewSpecName = llvm::formatv("{}_{}", specName, suffix).str();
+    ++suffix;
   }
```

Comment:
nit: You can turn this into a for loop since `suffix` is not used anywhere else

```suggestion
  for (unsigned suffix = 0; seenNames.contains(uniqueNewSpecName); ++suffix) {
    uniqueNewSpecName = llvm::formatv("{}_{}", specName, suffix).str();
  }
```